app/models/health/cha_saver.rb

```diff
@@ -37,31 +33,30 @@ def create
       end
     end
 
-
     def update
       @cha.class.transaction do
         @cha.completed_at = nil unless @complete
         @cha.save!
-        if @complete && @reviewed && @create_qa
-          @qualifying_activity.source_id = @cha.id
-          @qualifying_activity.save
+        if @complete && @reviewed && @create_qa && @cha.patient.recent_ssm_form.completed_at.present?
+          qualifying_activity = setup_qualifying_activity
+          qualifying_activity.save
         end
       end
     end
 
     protected def setup_qualifying_activity
       Health::QualifyingActivity.new(
         source_type: @cha.class.name,
+        source_id: @cha.id,
         user_id: @user.id,
         user_full_name: @user.name_with_email,
-        date_of_activity: Date.current,
+        date_of_activity: @cha.completed_at.to_date,
         activity: :cha,
-        follow_up: 'Implement Comprehensive Health Assessment',
+        follow_up: 'This writer completed CHA and SSM with patient.',
```

Comment:
This seems odd, the SSM could have bee done days before the CHA, and what happens if they don't occur in the right order?